assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * DashboardCTA component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import {
+	showErrorNotification,
+} from '../../../../../util';
+import GenericError from '../../../../../components/legacy-notifications/generic-error';
+import { CORE_MODULES } from '../../../../../googlesitekit/modules/datastore/constants';
+import { CORE_LOCATION } from '../../../../../googlesitekit/datastore/location/constants';
+import Button from '../../../../../components/Button';
+import Link from '../../../../../components/Link';
+import IdeaHubIcon from '../../../../../../svg/idea-hub.svg';
+import BulbIcon from '../../../../../../svg/bulb.svg';
+const { useSelect, useDispatch } = Data;
+
+const MODULE_SLUG = 'idea-hub';
+
+function DashboardCTA() {
+	const { connected, active } = useSelect( ( select ) => select( CORE_MODULES ).getModule( MODULE_SLUG ) );
+	const { activateModule } = useDispatch( CORE_MODULES );
+	const { navigateTo } = useDispatch( CORE_LOCATION );
+
+	const onClick = useCallback( async () => {
+		const { error, response } = await activateModule( MODULE_SLUG );
+
+		if ( ! error ) {
+			navigateTo( response.moduleReauthURL );
+		} else {
+			showErrorNotification( GenericError, {
+				id: `${ MODULE_SLUG }-setup-error`,
+				title: __( 'Internal Server Error', 'google-site-kit' ),
+				description: error.message,
+				format: 'small',
+				type: 'win-error',
+			} );
+		}
+	}, [ activateModule, navigateTo ] );
+
+	return (
+		<div className="googlesitekit-idea-hub-dashboardcta">
```

Comment:
@johnPhillips Can we rename the class name to `googlesitekit-idea-hub__dashboard--cta` ? This change applies to the other class names as well.

Reply:
@asvinb To be fair we don't have very clear precedent to follow regarding class names I think using `--` as suggested here would be another use that I don't think we have yet. Let's keep `--` to signify state/variant. In this case I think `googlesitekit-idea-hub__dashboard-cta` is probably the least inconsistent for a base 😆

Reply:
@aaemnnosttv I agree that we don't really have an agreed convention for class names but if we strictly follow the BEM methodologies, the class names can get very long and ugly. Maybe we can use a mix of our own methodology and BEM.  I'am thinking what's the best way to avoid having such class names in the code:

- googlesitekit-pagespeed-insights-web-vitals-metrics--field-data-unavailable
- googlesitekit-pagespeed-report-metric-value__display-value
- .googlesitekit-widget--analyticsAllTraffic__user-count-chart

Reply:
@asvinb and @aaemnnosttv What's the consensus, are we happy with `googlesitekit-idea-hub__dashboard-cta`?

Happy to go with whatever you think.

Reply:
I'll defer toe @aaemnnosttv for final decision 😁

Reply:
@asvinb @johnPhillips Let's not get too hung up on this since we don't have a clear rule for how we write our class names so whichever way we go, we'll still be inconsistent overall. IMO a utility-first approach avoids most of these problems to begin with which is my preferred "way to avoid having such class names in the code" 😄 We will need to revisit our strategy for CSS in the not too distant future so it's not worth spending too much time crafting the right name for.

Let's go with `googlesitekit-idea-hub__dashboard-cta` for now and we can work on defining the approach going forward (even if temporary) to reduce confusion in the future.